libraries/AC_AttitudeControl/AC_AttitudeControl.cpp

```diff
@@ -599,6 +538,135 @@ void AC_AttitudeControl::input_angle_step_bf_roll_pitch_yaw(float roll_angle_ste
     attitude_controller_run_quat();
 }
 
+// Command a Quaternion attitude with feedforward and smoothing
+void AC_AttitudeControl::input_thrust_vector_rate_heading(Vector3f thrust_vector, float heading_rate_cds)
+{
+    // Convert from centidegrees on public interface to radians
+    const float heading_rate = radians(heading_rate_cds * 0.01f);
+
+    // calculate the attitude target euler angles
+    _attitude_target.to_euler(_euler_angle_target.x, _euler_angle_target.y, _euler_angle_target.z);
+
+    thrust_vector.normalize();
```

Comment:
I think we need to add a divide by zero check here because thrust_vector could theoretically at least be {0,0,0}

Reply:
Done